cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java

```diff
@@ -123,17 +165,29 @@ public void revoke(EntityId entityId, Principal principal, Set<Action> actions)
           String.format("The given principal '%s' is of unsupported type '%s'.", principal.getName(),
                         principal.getType()));
     }
+    loadCache(entityId, principal, actions, false);
```

Comment:
Why load the cache on grant/revoke ? Do we expect enforce call to come right after in every scenario ? I think the cache should only be loaded at one place i.e. enforce. These guys should only invalidate.